kivy/core/text/markup.py

```diff
@@ -775,5 +816,10 @@ def p_restricted(line, uw, c):
         lw = sum([word.lw for word in line1])
         lh = max([word.lh for word in line1]) * line_height
         self.options = old_opts
-        return lw + 2 * xpad, lh + 2 * ypad, [LayoutLine(0, 0,
-            lw, lh, 1, 0, line1)]
+        if uw < sum([word.lw for word in line]):
```

Comment:
Again, this computation is already done. Why do it twice? Also, shouldn't it be line1?